### PR TITLE
Ported over  publish_prod.sh release script

### PR DESCRIPTION
**Pull request by zARODz11z**
### What does this PR do?
This PR ports over a publish_prod.sh script to publish the datadog-cdk-constructs to NPM  as well as bumps the version on github.

### Motivation
The datadog-cdk-constructs have not yet been published to NPM, NPM should be a better experience for getting access to the datadog-cdk-constructs as opposed to cloning the repository from github. In the future this script should make the process of releasing new versions of the cdk constructs easier.

### Testing Guidelines
I am currently researching how to write unit tests for a bash shell script but am currently unsure if any of those frameworks are permitted such as [shunit2](https://github.com/kward/shunit2).

### Additional Notes
The script is nearly identical to the publish_prod.sh script from the Datadog Serverless plugin with minor changes to not updating the layer versions because the Datadog CDK constructs seems to create the layer ARNs on the fly.

### Types of Changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

**Review by DarcyRaynerDD — COMMENTED** [review]
This looks good. It might also make sense to port over the .npmignore file. It will force the yarn publish step to only upload specific files. In the datadog-lambda-js version, you can see it ignores every file using a wildcard '*', then excludes the files it wants to keep from being ignored. https://github.com/DataDog/datadog-lambda-js/blob/main/.npmignore

**Comment by zARODz11z** [comment]
> This looks good. It might also make sense to port over the .npmignore file. It will force the yarn publish step to only upload specific files. In the datadog-lambda-js version, you can see it ignores every file using a wildcard '*', then excludes the files it wants to keep from being ignored. https://github.com/DataDog/datadog-lambda-js/blob/main/.npmignore

@DarcyRaynerDD Currently the Datadog CDK constructs has a npmignore with the following inside:
&ast;.ts
 !*.d.ts
 .cdk.staging
 cdk.out

Seems like all we want to exclude all typescript files except ones ending in d.ts. Then we also want to exclude files associated with the CDK deployment.

File: scripts/publish_prod.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+if [ -z "$AWS_SECRET_ACCESS_KEY" ]; then
+    echo 'AWS_SECRET_ACCESS_KEY not set. Are you using aws-vault?'
+    exit 1
+fi
+
+if [ -z "$AWS_SESSION_TOKEN" ]; then
+    echo 'AWS_SESSION_TOKEN not set. Are you using aws-vault?'
+    exit 1
+fi
```
Comment (nhinsch) [inline scripts/publish_prod.sh:29]: Do we need AWS credentials to publish to NPM?

File: scripts/publish_prod.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+yarn version --new-version "$VERSION"
+
+echo 'Publishing to Node'
+yarn test
+yarn build
+yarn publish --new-version "$VERSION"
+
+echo 'Pushing updates to github'
+git push origin master
+git push origin "refs/tags/v$VERSION"
```
Comment (nhinsch) [inline scripts/publish_prod.sh:64]: I would add a reminder to the user to add the release notes in GitHub

**Review by nhinsch — APPROVED** [review]
Looks good, I left a couple of comments

**Pull request merged**